woocommerce/woocommerce
# Use syncpack customTypes to keep pnpm and node in sync. Update pnpm to 8.6.7

### Changes proposed in this Pull Request:

In past PRs to update pnpm engine we had to use find/replace, but syncpack supports any custom field on package json via `customTypes`.

This uses `customTypes` to keep node engine and pnpm engine in sync and also updates pnpm to latest version.

Other small changes:

* Tidied up the filter regex to add node and pnpm to the filter. Filter had bugs, ie it didn't respect @typescript-eslint/** properly.

### How to test the changes in this Pull Request:


1. Personally I don't think there is much to test here, CI passing and building should be acceptable
2. However you could have a play with the filter regex to test it if you want:
3. Any `versionGroup` that does not have `isIgnored` you can try changing the version on it
4. Then see if the filter works for that group by running from monorepo root: `pnpm syncpack list-mismatches`
5. It should show mismatches. I tested this for wild cards and some of the dependencies that are exact matched in the filter, including checking that `node` would work if I changed the version to something else.



### Changelog entry




<details>

#### Significance




#### Type




#### Message
Update pnpm to 8.6.7

#### Comment 

</details>
